### DISCUSSION TITLE
Override response status code

### DISCUSSION THREAD

**Discussion by vnagalingam** [graphql/graphql-http#59] [Q&A] [answered]
Hi, For errors, how can I format error message and change the response status code? With express-graph, there was an option to set customFormatErrorFn. Is there anything similar in graphql-http?

Thanks

**Comment by enisdenjo** [comment, marked as answer]
Hey there, starting from [v1.18.0](https://github.com/graphql/graphql-http/releases/tag/v1.18.0), you can use the [`formatError` handler option](https://github.com/graphql/graphql-http/blob/main/docs/interfaces/handler.HandlerOptions.md#formaterror) to format errors to your requirement.

On the other note, status codes shouldn't be changed when dealing with handled errors since graphql-http wants to respect the GraphQL over HTTP spec. However, for anything requiring custom responses, you can use the hooks return value - for example handling auth:

```js
import { createHandler } from 'graphql-http';
import { checkAuth } from './my-auth';

const handler = createHandler({
  async onSubscribe(req) {
    const isAuthorized = await checkAuth(req);
    if (!isAuthorized) {
      return ['Who are you?', { status: 401, statusText: 'Unauthorized' }];
    }
  },
});
```
